### DISCUSSION TITLE
How to Insert data from an API into supabase DB

### DISCUSSION THREAD

**Discussion by ChrisB007** [supabase/supabase#5892] [Questions]
Hey guys, can you help me figure this out - I'm a supabase newbie. So, I am trying to save data that I fetched from an api to my supababse database (using Nextjs), but can't tell where to plug in my supabase related code to 'Insert' the data. Here is what my code looks like

```export async function getServerSideProps({ req }) {
  //Base Url
  const gBaseUrl = 'https://www.googleapis.com/youtube/v3';

  //Supabase related
  //Insert many rows

  // const { data, error } = await supabase
  // .from('table')
  // .insert([
  //   { some_column: 'someValue' },
  //   { some_column: 'otherValue' },
  // ])

  const channelID = [];

  for (const channel of channelList) {
    const channelData = await fetch(
      `${gBaseUrl}/channels?part=snippet,contentDetails,statistics&id=${channel.channelID}&key=${process.env.YOUTUBE_API_KEY}`
    ).then((res) => res.json());
    channelID.push(channelData.items); // Need to save "channelData.items" coming from YoutubeAPI to the database.
  }

  //Return data
  return {
    props: {
      channelID,
    },
  };
}
```

How can I accomplish this?

Thanks in advance

**Comment by CrazyMarvin** [comment]
> Hey guys, can you help me figure this out - I'm a supabase newbie. So, I am trying to save data that I fetched from an api to my supababse database (using Nextjs), but can't tell where to plug in my supabase related code to 'Insert' the data. Here is what my code looks like
> 
> ```export async function getServerSideProps({ req }) {
>   //Base Url
>   const gBaseUrl = 'https://www.googleapis.com/youtube/v3';
> 
>   //Supabase related
>   //Insert many rows
> 
>   // const { data, error } = await supabase
>   // .from('table')
>   // .insert([
>   //   { some_column: 'someValue' },
>   //   { some_column: 'otherValue' },
>   // ])
> 
>   const channelID = [];
> 
>   for (const channel of channelList) {
>     const channelData = await fetch(
>       `${gBaseUrl}/channels?part=snippet,contentDetails,statistics&id=${channel.channelID}&key=${process.env.YOUTUBE_API_KEY}`
>     ).then((res) => res.json());
>     channelID.push(channelData.items); // Need to save "channelData.items" coming from YoutubeAPI to the database.
>   }
> 
>   //Return data
>   return {
>     props: {
>       channelID,
>     },
>   };
> }
> ```
> 
> How can I accomplish this?
> 
> Thanks in advance

I would like to do something similar.

I am using data from [The Movie Database (TMDB)](https://www.themoviedb.org/documentation/api) which I would like to put into my Supabase so that I don't put to much "pressure" on TMDB's server.

The project [sneak.page](https://sneak.page/) is [open source](https://github.com/Crazy-Marvin/sneak.page/) and you may see the regarding [issue](https://github.com/Crazy-Marvin/sneak.page/issues/14).

Is there a solution within Supabase which could be used for this like [Edge Functions](https://supabase.com/edge-functions) or a direct command in the database? 🤔

Thanks for any help.

**Comment by CrazyMarvin** [comment]
Supabase support recommends to use [Supabse Edge Functions](https://supabase.com/edge-functions) in combination with [Bulk Loading](https://supabase.com/docs/guides/database/tables#bulk-data-loading).
